client/X11/xf_rail.c

```diff
@@ -880,6 +907,35 @@ static BOOL xf_rail_window_cached_icon(rdpContext* context,
 static BOOL xf_rail_notify_icon_common(rdpContext* context,
                                        WINDOW_ORDER_INFO* orderInfo, NOTIFY_ICON_STATE_ORDER* notifyIconState)
 {
+	xfContext* xfc = NULL;
+	xfAppNotifyIcon* notifyIcon = NULL;
+	xfc = (xfContext*) context;
+
+	if (orderInfo->fieldFlags & WINDOW_ORDER_STATE_NEW)
+	{
+		notifyIcon = (xfAppNotifyIcon*) calloc(1, sizeof(xfAppNotifyIcon));
+		notifyIcon->xfc = xfc;
+		notifyIcon->windowId = orderInfo->windowId;
+		notifyIcon->notifyIconId = orderInfo->notifyIconId;
+
+		if (!notifyIcon)
+			return FALSE;
+
+		HashTable_Add(xfc->railNotifyIcons, (void*)(UINT_PTR) orderInfo->notifyIconId,
```

Comment:
'0' is a valid notification ID but `HashTable_Add` will not accept that.

Reply:
Does that situation that '0' as a notification ID exist in practical ? How do I suppose to fix this ? :sweat_smile:

Reply:
yes it does. @hardening pointed me in a simple direction, just set a comparison function that compares the pointers by type (UINT32) and use something like `&notifyIcon->notifyIconId` as key